SPECS/kpatch/mariner_update.patch

```diff
@@ -0,0 +1,153 @@
+From 02d1b897968b6d01d8c38c4ad64120f81fa53247 Mon Sep 17 00:00:00 2001
+From: Pawel Winogrodzki <pawelwi@microsoft.com>
+Date: Wed, 15 Jun 2022 10:51:00 -0700
+Subject: [PATCH] Initial adjustments for CBL-Mariner.
+
+---
+ kpatch-build/kpatch-build | 74 +++++++++++++++++++++++----------------
+ 1 file changed, 43 insertions(+), 31 deletions(-)
+
+diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
+index 634095d..8868dbc 100755
+--- a/kpatch-build/kpatch-build
++++ b/kpatch-build/kpatch-build
+@@ -205,21 +205,6 @@ rhel_kernel_version_gte() {
+         [  "${ARCHVERSION}" = "$(echo -e "${ARCHVERSION}\\n$1" | sort -rV | head -n1)" ]
+ }
+ 
+-# klp.arch relocations were supported prior to v5.8
+-# and prior to 4.18.0-240.el8
+-use_klp_arch()
+-{
+-	if kernel_is_rhel; then
+-		! rhel_kernel_version_gte 4.18.0-240.el8
+-	else
+-		! kernel_version_gte 5.8.0
+-	fi
+-}
+-
+-rhel_kernel_version_gte() {
+-        [  "${ARCHVERSION}" = "$(echo -e "${ARCHVERSION}\\n$1" | sort -rV | head -n1)" ]
+-}
+-
+ # klp.arch relocations were supported prior to v5.8
+ # and prior to 4.18.0-284.el8
+ use_klp_arch()
+@@ -638,13 +623,23 @@ if [[ $DEBUG -eq 1 ]] || [[ $DEBUG -ge 3 ]]; then
+ 	set -o xtrace
+ fi
+ 
++# Don't check external file.
++# shellcheck disable=SC1090
++if [[ -z "$USERSRCDIR" ]] && [[ -f "$RELEASE_FILE" ]]; then
++	source "$RELEASE_FILE"
```

Comment:
Please add a runtime requirement to `kpatch-build` on `mariner-release`, now that the package is no longer present in the build chroot by default.

Reply:
Ah, nice catch, thanks!